src/Bridge/Doctrine/MongoDbOdm/Extension/PaginationExtension.php

```diff
@@ -64,12 +64,18 @@ public function applyToCollection(Builder $aggregationBuilder, string $resourceC
             throw new RuntimeException(sprintf('The repository for "%s" must be an instance of "%s".', $resourceClass, DocumentRepository::class));
         }
 
+        $resultsAggregationBuilder = $repository->createAggregationBuilder()->skip($offset);
+        if ($limit > 0) {
+            $resultsAggregationBuilder->limit($limit);
+        } else {
+            // Results have to be 0 but MongoDB does not support a limit equals to 0.
+            $resultsAggregationBuilder->match()->field('___')->equals('limit0');
```

Comment:
maybe a constant in the ` src/Bridge/Doctrine/MongoDbOdm/Paginator.php` for the strings?